### ISSUE TITLE
Style/DoubleNegation: does not register an offence for last line of a returned hash/array

### ISSUE THREAD

**Issue by djfpaagman** [rubocop/rubocop#10846]
## Expected behavior
	
It should detect all cases when used inside a hash or array.

## Actual behavior

The Style/DoubleNegation cop does not detect double negations for the last line of a hash or array whe it is the returned value of a method. I think this is because of the `allowed_in_returns` setting, but I'm not entirely sure.

https://github.com/rubocop/rubocop/blob/7dee99a63db2eee15a188c26f7bf30a6013c2bfb/lib/rubocop/cop/style/double_negation.rb#L8-L10

## Steps to reproduce the problem

Simplified examples:

```ruby
# frozen_string_literal: true

def multi_line_hash
  {
    foo: !!foo,
    bar: !!bar
  }
end

def multi_line_array
  [
    !!foo,
    !!bar
  ]
end
```

Only detects the offense on the lines for `foo`:

```
test.rb:5:10: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    foo: !!foo,
         ^
test.rb:12:5: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    !!foo,
    ^
```
--- 

```ruby
# frozen_string_literal: true

def singe_line_hash
  { foo: !!foo }
end

def single_line_array
  [!!foo]
end
```

Does not detect an offense at all:

```
1 file inspected, no offenses detected
```

---

```ruby
# frozen_string_literal: true

def hash_not_directly_returned
  hash = {
    foo: !!foo,
    bar: !!bar
  }

  do_something_with(hash)
end

def array_not_directly_returned
  array = [
    !!foo,
    !!bar
  ]

  do_something_with(array)
end

```

Correctly detects the offense for all lines:

```
test.rb:5:10: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    foo: !!foo,
         ^
test.rb:6:10: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    bar: !!bar
         ^
test.rb:14:5: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    !!foo,
    ^
test.rb:15:5: C: [Correctable] Style/DoubleNegation: Avoid the use of double negation (!!).
    !!bar
    ^
```

## RuboCop version

```
❯ rubocop -V
1.32.0 (using Parser 3.1.2.0, rubocop-ast 1.19.1, running on ruby 3.0.3 arm64-darwin21)
```
